templates/create-daml-app/ui/src/config.ts

```diff
@@ -14,10 +14,13 @@ export const deploymentMode: DeploymentMode =
   ? DeploymentMode.PROD_DABL
   : DeploymentMode.PROD_OTHER;
 
-export const ledgerId =
+// Decide the ledger ID based on the deployment mode first,
+// then an environment variable, falling back on the sandbox ledger ID.
+export const ledgerId: string =
   deploymentMode === DeploymentMode.PROD_DABL
   ? window.location.hostname.split('.')[0]
-  : 'create-daml-app-sandbox';
+  : process.env.REACT_APP_LEDGER_ID
```

Comment:
Did you test this? The documentation for `dotenv` says that you have to require and configure it via `require('dotenv').config()` see https://www.npmjs.com/package/dotenv. I’m struggling to see how the `.env` file is magically picked up otherwise.

Reply:
I did test it, simply with a `console.log()` of the env var in the App component. The `.env` file is magically picked up, I'm not sure where it is configured but it must be somewhere in the create react app setup.

Reply:
Cool, thanks for confirming :+1: